node/tests/SharedTests.ts

```diff
@@ -316,6 +316,39 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `msetnx test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key1 = "{key}-1" + uuidv4();
+                const key2 = "{key}-2" + uuidv4();
+                const key3 = "{key}-3" + uuidv4();
+                const nonExistingKey = uuidv4();
+                const value = uuidv4();
+                const keyValueMap1 = {
+                    [key1]: value,
+                    [key2]: value,
+                };
+                const keyValueMap2 = {
+                    [key2]: value,
+                    [key3]: value,
+                };
+
+                expect(await client.msetnx(keyValueMap1)).toEqual(true);
+
+                checkSimple(
+                    await client.mget([key1, key2, nonExistingKey]),
+                ).toEqual([value, value, null]);
+
+                expect(await client.msetnx(keyValueMap2)).toEqual(false);
+
+                expect(await client.get(key3)).toEqual(null);
+                checkSimple(await client.get(key2)).toEqual(value);
```

Comment:
Can you please add a test passing an empty map to msetnx and asserting that a RequestError is thrown?